swap/src/bin/swap.rs

```diff
@@ -336,51 +337,79 @@ async fn init_monero_wallet(
     Ok((monero_wallet, monero_wallet_rpc_process))
 }
 
-async fn determine_btc_to_swap(
-    request_quote: impl Future<Output = Result<BidQuote>>,
-    initial_balance: impl Future<Output = Result<bitcoin::Amount>>,
+async fn determine_btc_to_swap<FB, TB, FMG, TMG, FS, TS>(
+    bid_quote: BidQuote,
+    mut current_maximum_giveable: bitcoin::Amount,
     get_new_address: impl Future<Output = Result<bitcoin::Address>>,
-    wait_for_deposit: impl Future<Output = Result<bitcoin::Amount>>,
-    max_giveable: impl Future<Output = Result<bitcoin::Amount>>,
-) -> Result<bitcoin::Amount> {
-    debug!("Requesting quote");
-
-    let bid_quote = request_quote.await?;
-
-    info!("Received quote: 1 XMR ~ {}", bid_quote.price);
-
-    // TODO: Also wait for more funds if balance < dust
-    let initial_balance = initial_balance.await?;
+    balance: FB,
+    max_giveable: FMG,
+    sync: FS,
+) -> Result<(bitcoin::Amount, bitcoin::Amount)>
+where
+    TB: Future<Output = Result<bitcoin::Amount>>,
+    FB: Fn() -> TB,
+    TMG: Future<Output = Result<bitcoin::Amount>>,
+    FMG: Fn() -> TMG,
+    TS: Future<Output = Result<()>>,
+    FS: Fn() -> TS,
+{
+    let max_giveable = if current_maximum_giveable == bitcoin::Amount::ZERO
+        || current_maximum_giveable < bid_quote.min_quantity
+    {
+        let deposit_address = get_new_address.await?;
+        let minimum_amount = bid_quote.min_quantity;
+        let maximum_amount = bid_quote.max_quantity;
 
-    let balance = if initial_balance == Amount::ZERO {
         info!(
-            "Please deposit the BTC you want to swap to {} (max {})",
-            get_new_address.await?,
-            bid_quote.max_quantity
+            %deposit_address,
+            %current_maximum_giveable,
```

Comment:
Not sure we want to expose this to the user, but I see the CLI as a temporary solution so I think it is fine. 
We could also opt for naming it different as in `maximum_swap_amount` and potentially always print `amount` and `fees` together within here.